python_modules/dagster/dagster/_cli/api.py

```diff
@@ -79,6 +84,47 @@ def send_to_buffer(event):
                 sys.exit(return_code)
 
 
+def _is_isolated_run(dagster_run: DagsterRun) -> bool:
+    return dagster_run.tags.get("dagster/isolation") != "disabled"
+
+
+def _truthy_tag_value(dagster_run: DagsterRun, tag: str, default: str = "false") -> bool:
+    return dagster_run.tags.get(tag, default).casefold() in ["true", "1", "yes", "on"]
+
+
+def _should_start_metrics_thread(dagster_run: DagsterRun) -> bool:
+    return _truthy_tag_value(dagster_run, "dagster/run_metrics")
+
+
+def _enable_python_runtime_metrics(dagster_run: DagsterRun) -> bool:
+    return _truthy_tag_value(dagster_run, "dagster/python_runtime_metrics")
+
+
+def _report_container_metrics_as_engine_events(dagster_run: DagsterRun) -> bool:
+    return _truthy_tag_value(dagster_run, "dagster/container_metrics_engine_events")
```

Comment:
is there value in enabling these individually? vs. a global "collect_run_metrics" tags? seems simpler to describe in the docs

Reply:
1. `python_runtime_metrics`: this was more of a cheap shot in the dark in case we have to debug strange GC perf issues like I had to in the past where the java runtime gc metrics were invaluable. but we might never run into this so keeping it separate makes sense to me.

2. `container_metrics_engine_events`, for the same reason you have reservation about perf impact and usability of this, I don't think I want to keep the engine event reporting feature when this is mature. But this was a great way to iterate faster on developing this or possibly to give some visibility to the end user until the UI is built.

Reply:
engine events reporting is removed - will keep python_runtime_metrics unless you object